### PR TITLE
chose tables in 'cdm_apply_filters()'

### LINKED ISSUE #126: apply filters only to selected tbls?
Maybe we could support a second argument to `cdm_apply_filters()`, giving the possibility to specify which tables we want to use `tbl()` for. Might save time in some situations (only interested in effect of filtering table "a" on table "b", but not other tables).

### PR DESCRIPTION
**Pull request by TSchiefer**
- new argument `...` to `cdm_apply_filters()` to specify, which tables should be filtered
- tests

closes #126

### CODE REVIEW

File: R/filter-dm.R [hunk 1]
```diff
@@ -89,13 +91,16 @@ set_filter_for_table <- function(dm, table, filter_exprs, zoomed) {
...
   check_not_zoomed(dm)
-  def <- cdm_get_def(dm)
-
-  def$data <- map(def$table, ~ tbl(dm, .))
-
-  cdm_reset_all_filters(new_dm3(def))
+  vars <- tidyselect_table_names(dm)
+  selected <- tidyselect::vars_select(vars, ...)
+  # in case of empty ellipsis all tables should be in selection
+  if (is_empty(selected)) selected <- names(dm)
```
Comment (krlmlr) [inline R/filter-dm.R]: Only `if (missing(...))` (before even calling tidyselect)?

File: R/filter-dm.R [hunk 2]
```diff
@@ -22,14 +22,16 @@
...
-#' @param ... Logical predicates defined in terms of the variables in `.data`, passed on to [dplyr::filter()].
+#' @param ... For `cdm_filter()`: Logical predicates defined in terms of the variables in `.data`, passed on to [dplyr::filter()].
 #' Multiple conditions are combined with `&` or `,`. Only rows where the condition evaluates
 #' to TRUE are kept.
 #'
 #' The arguments in ... are automatically quoted and evaluated in the context of
 #' the data frame. They support unquoting and splicing. See `vignette("programming", package = "dplyr")`
 #' for an introduction to these concepts.
 #'
+#' For `cdm_apply_filters()`: Unquoted names of the tables to apply the filters to.
```
Comment (krlmlr) [inline R/filter-dm.R]: Can we split documentation files?

File: R/filter-dm.R [hunk 3]
```diff
@@ -89,13 +91,16 @@ set_filter_for_table <- function(dm, table, filter_exprs, zoomed) {
...
-
-  def$data <- map(def$table, ~ tbl(dm, .))
-
-  cdm_reset_all_filters(new_dm3(def))
+  vars <- tidyselect_table_names(dm)
+  selected <- tidyselect::vars_select(vars, ...)
+  # in case of empty ellipsis all tables should be in selection
+  if (is_empty(selected)) selected <- names(dm)
+  new_def <- cdm_get_def(dm) %>%
+    mutate(data = map(table, ~if_else(.x %in% selected, list(tbl(dm, .x)), data[.x == table])) %>% flatten)
```
Comment (krlmlr) [inline R/filter-dm.R]: `def$data[selected] <- map(def$data[selected], ...)`

File: R/filter-dm.R [hunk 4]
```diff
@@ -89,13 +91,16 @@ set_filter_for_table <- function(dm, table, filter_exprs, zoomed) {
...
+
+  cdm_reset_all_filters(new_dm3(new_def))
```
Comment (krlmlr) [inline R/filter-dm.R]: Is this still correct?

Comment (TSchiefer) [reply R/filter-dm.R]: Very valid point. In my opinion, it should be still correct and we should add something like the following to the documentation: "A call to `cdm_apply_filters()` resets all filter conditions that are stored in the `dm`, independent if the filters are only applied to a subset of the tables"

But I'm open to suggestions

Comment (krlmlr) [reply R/filter-dm.R]: I forgot about the duality here. In this case let's shelve this for now until we have a better understanding of filtering and applying filters.

Comment (TSchiefer) [reply R/filter-dm.R]: It somehow doesn't seem very likely to me, that we're going to move away far from our current API regarding the filtering, apart from #184, which doesn't affect this issue though. I might be wrong, but it seems to me that it's just a question of deciding whether we want the filter conditions still to be stored or dropped after `cdm_apply_filters()`

**Comment by krlmlr** [comment]
Let's shelve for now.

**Pull request closed without merging**